### ISSUE TITLE
Standard text font does not support latin characters.

### ISSUE THREAD

**Issue by GAZ082** [naturalcrit/homebrewery#116]
Hello there. Beautiful piece of software you've made. But have an issue, apparently, the font used as standard text does not have latin characters (such as áéíóú, ñ, ¡,¿) and instead renders the fallback font, which is noticeably bigger.

Dont know if something can be made, as this is a thing the font maker should address, but hey.

Thanks!

**Comment by stolksdorf** [comment]
Sorry about your issue! You are right, there's not a ton I can do. I have a font guy I'll chat with and see how difficult it would be to add. Thanks for the report!

**Comment by GAZ082** [comment]
LOL, you replied so fast i thought it was an auto response, haha. Thank you man.

**Comment by stolksdorf** [comment]
The Homebrewery uses 4 fonts: Main Serif, main Sans serif, headers, and the fancy first letter. Which ones do latin characters not work on?

Also, summoning @mimosacb to help out with fonts.

**Comment by mimosacb** [comment]
I'll take a look at the source font files later today, but if the glyphs aren't included the best we can do is replace the font with one that has them. Next best thing would be to find a fallback with similar metrics.

**Comment by GAZ082** [comment]
It's a mix. The heading font is ok but lacks the (¡), odd, since it has the (¿)
The normal font is the one with the less features. I made a document for you to check!

http://www.naturalcrit.com/homebrew/edit/BklarITFm

I think this also would be a good opportunitiy to work more on the fonts, ie, i forsee in a near future some germanic speaker (german, swedish, norwegian, etc.) asking for some special symbols.

Thanks!

**Comment by Idrinth** [comment]
@mimosacb there's quite a few words starting with Ä, Ö and Ü in german, some of them could easily start a paragraph or have an upper case letter in the headers(all nouns start upper-case).

Magische Öle und ähnliche Flüssigkeiten (magic oils and similar liquids) 
Would work with an alchemist-like class for example, ends up with a number instead of the Ö.

**Comment by calculuschild** [comment]
The font currently used for the standard paragraph text is "BookInsanity", an open-source font created by Reddit user Solbera, based on the commercial font used by Wizards of the Coast, "BookMania". Use of the open-source avoids some nebulous legal and copyright issues if users want to sell their Homebrewery creations.

However, it has come to my attention that some other Reddit users have fixed and updated the BookInsanity fonts and they appear to include these additional latin characters. I'm going to test the updated BookInsanity font and maybe we can finally fix this issue.

**Comment by calculuschild** [comment]
Another Reddit user posted some tweaks to better position the capitalized letters in the Header fonts:

https://www.reddit.com/r/homebrewery/comments/kjoo29/solbera_imitation_tweak/

**Comment by AlexeySachkov** [comment]
There is a repo where several Solbera's fonts are stored and maintained (I hope so): https://github.com/jonathonf/solbera-dnd-fonts

**Comment by calculuschild** [comment]
Yep there's the ones I found earlier. Thanks for linking it up. This should be a pretty easy fix if someone wants to make a PR for it (I'm focusing on some other items at the moment).

**Comment by ericscheid** [comment]
See also https://www.reddit.com/r/homebrewery/comments/le2r98/

**Comment by calculuschild** [comment]
Reddit user who added accents and fixed some kerning issues on most of the fonts. https://www.reddit.com/r/UnearthedArcana/comments/71wzc2/new_indesign_template_and_open_source_fonts/

**Comment by Gazook89** [comment]
> There is a repo where several Solbera's fonts are stored and maintained (I hope so): https://github.com/jonathonf/solbera-dnd-fonts

This repo just updated their fonts with the updated ones from that Reddit thread

**Comment by ericscheid** [comment]
Something weird is happening with the MrJeeves font used for brew titles on `/user/name` page vs what appears on the brew page. I can't see where the CSS is different.

![image](https://user-images.githubusercontent.com/2605467/111174069-3fe21a00-85fb-11eb-8e40-683f8b27d5e4.png)

![image](https://user-images.githubusercontent.com/2605467/111173980-29d45980-85fb-11eb-8c31-46fea39b9beb.png)

**Comment by calculuschild** [comment]
@ericscheid You have different characters in the two examples, ė and é . é is supported in the font but the other is not.

**Comment by ericscheid** [comment]
> @ericscheid You have different characters in the two examples, ė and é . é is supported in the font but the other is not.

Oooh, well spotted. Though I now have no idea how &edot; got there — there is no handy or accidental keypress in Mac OS to type that. 

`&edot;` not being supported in the font is (mostly) fine, it isn't a common diacritic. We might upset the Lithuanians, but that's about it.

**Comment by calculuschild** [comment]
Closing! This has been added to the V3 renderer via #1267

**Comment by ericscheid** [comment]
Re-opening since some characters are still missing umlautes.

Some of those caps with diaeresis are not in common language use, but Ö is a specific example [called out above](https://github.com/naturalcrit/homebrewery/issues/116#issuecomment-252374389).

![some caps missing umlautes](https://user-images.githubusercontent.com/2605467/147318876-0446f96d-c6ae-41fc-bb74-04bc7a791494.png)

**Comment by calculuschild** [comment]
A Reddit post with details on some other possibly missing characters. Link to example brew: https://homebrewery.naturalcrit.com/share/_UFl9V3IYuZ8

**Comment by Gazook89** [comment]
Is there a 'finish point' for this issue such that it can be closed?  If so, what is it?  A glyph for *n* unicode characters on every default font included with Homebrewery?  

[Glyphr Studio](https://www.glyphrstudio.com/online/) looks like a promising free tool for doing some font work as long as we don't need lots of different settings....Since @calculuschild seems to be at least somewhat familiar with font editing, could you take a look at Glyphr and determine if it will *work* for what needs to be done on this Issue or if it's going to only get us halfway?  

If Glyphr will work, and we can set a series of milestones and guidelines, perhaps this issue can be banged out fairly quickly amongst a few users.  Something like:

## Milestones
Complete glyphs outlined on [this brew](https://homebrewery.naturalcrit.com/share/_UFl9V3IYuZ8) for these fonts:
- [ ] Booksanity (body font)
- [ ] Nodesto
- [ ] Scaly Sans
- [ ] ....

I would volunteer to do this but just don't want to have to do it twice due to misunderstanding some small thing.  And if it's enough to just get the Ö done, then great.

**Comment by 5e-Cleric** [comment]
I have worked with Glyphr studio, and can guarantee it can very much help with this issue, what kind of milestones would you suggest?

**Comment by 5e-Cleric** [comment]
Going to use [this brew](https://homebrewery.naturalcrit.com/share/niFq-y3XB0g7) to test fonts from now on, and i suggest we stick to Basic Latin + Latin-1 Supplement.

**Comment by dbolack-ab** [comment]
Which packup of the fonts are we using?  https://github.com/lucastucious/solbera-dnd-fonts ?

**Comment by calculuschild** [comment]
Most of them originate from there. However we have since added several missing characters, fixed character spacing/alignment, etc. The ones we modified should have "Remake" in the title.

**Comment by dbolack-ab** [comment]
To resolve this issue, we need:

+ Someone(s) who does this regularly enough to know the missing pieces

or

- A list of missing glyphs 
- The last edited versions of the faces

Or we should close this as a can't/won't fix.
